src/sentry/api/authentication.py

```diff
@@ -69,6 +71,38 @@ def authenticate_credentials(self, userid, password):
         return (AnonymousUser(), key)
 
 
+class ClientIdSecretAuthentication(QuietBasicAuthentication):
+    """
+    Authenticates a Sentry Application using its Client ID and Secret
+
+    This will be the method by which we identify which Sentry Application is
+    making the request, for any requests not scoped to an installation.
+
+    For example, the request to exchange a Grant Code for an Api Token.
+    """
+
+    def authenticate(self, request):
+        data = json.loads(request.body)
+        client_id = data.get('client_id')
+        client_secret = data.get('client_secret')
+
+        if client_id is None or client_secret is None:
```

Comment:
What about `''`, 0 and `[]`?

Reply:
Yea, good call, should probably handle those as well.

Reply:
All in due time......